go/master/client.go

```diff
@@ -33,6 +46,46 @@ func NewClient(addr Addresser, bufSize int) *Client {
 	return c
 }
 
+// NewEtcdClient create a new master client by etcd
+//
+// etcdEndpoints is the endpoints for etcd, it's separated by "," such as
+// "172.0.1.0:2379,172.0.1.1:2379"
+// bufSize is the record buffer size. NextRecord will read from this buffer.
+func NewEtcdClient(etcdEndpoints string, etcdTimeout int, bufSize int) *Client {
+	timeout := time.Second * time.Duration(etcdTimeout)
+	ep := strings.Split(etcdEndpoints, ",")
+	cli, err := clientv3.New(clientv3.Config{
+		Endpoints:   ep,
+		DialTimeout: timeout,
+	})
+	if err != nil {
+		log.Errorf("Init etcd connection failed: %v", err)
+		panic(err)
+	}
+	log.Debugf("Connected to etcd: %s\n", etcdEndpoints)
+	for {
+
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		resp, err := cli.Get(ctx, masterAddrPath)
+		cancel()
+		if err != nil {
+			log.Errorf("Fetch master addr failed, %v\n", err)
+			time.Sleep(timeout)
+			continue
+		}
+		kvs := resp.Kvs
+		if len(kvs) == 0 {
+			log.Infoln("Waiting for master be ready ...\n")
+			time.Sleep(timeout)
+			continue
+		}
+
+		mAddr := kvs[0].Value
+		log.Debugf("Fetched master address: %s\n", mAddr)
+		return NewClient(MasterAddresser(mAddr), bufSize)
```

Comment:
Could you take a look at this piece of code: https://github.com/PaddlePaddle/Paddle/blob/develop/go/master/client.go#L80

The idea is the client takes a `Addresser` and query it to know if it have changed.
This current implementation is create a "constant" `Addresser` that returns the same address every time. The thing is master server could be killed and restarts with a different address.

So I think a better approach could be this etcd query function be the implementation of the `Addresser` interface. So the master server will know the updated address when calling `Address`.

Another detail is when implementing the `Addresser`, it don't need to create a client in every call to `Address`, it should reuse the same client.